### ISSUE TITLE
Presto/Spark Support - .prestoSchema creation and Spark writes appear to require md5 support

### ISSUE THREAD

**Issue by motorman-ibm** [noobaa/noobaa-core#6228]
### Environment info
- NooBaa Version: ns-fs
- Platform: Kubernetes 1.19.x
Presto database application

### Actual behavior
When running Presto with noobaa
1. Using the appropriately defined noobaa catalog
2. Client "create schema" fails (worked around by manually creating subdir and copying .prestoSchema into subdir)
3. Client "create table" fails - appears to be md5 error

### Expected behavior
1. Using the appropriately defined noobaa catalog
2. At the client  create a schema
3. Using the schema create a table

### Steps to reproduce
1. Install presto and noobaa
2. create schema
3. create table

### More information - Screenshots / Logs / Other output

Comments - I understand this is more than one issue.  I believe that the initial issues are covered by other issues
- subdirectories not created by "make bucket"
- deep listing may be required (the workaround) 

Which leaves 2 potential issues from this submission
- md5 support (this may be tricky - I don't think presto uses this for all objects since we have run previously in a pre-created container)
- extra data in .prestoSchema file - this could be an artifact of my workaround to allow Presto to progress to the md5 error - I'm not claiming that is an issue - just posting for awareness in case it turns out to be one

[presto_and_noobaa_issues.pdf](https://github.com/noobaa/noobaa-core/files/5410802/presto_and_noobaa_issues.pdf)

**Comment by motorman-ibm** [comment]
The requirement **might** not be to implement md5 on all object data, just on puts.
As reference see the following:
https://oak-tree.tech/blog/spark-kubernetes-minio
```enabling "fast" uploads, which tells Spark that MinIO will not be returning an MD5 hash of the file contents```
In my testing, even with fast upload enabled, the md5 error was still issued by Spark - that leads me to believe (along with the moniker "upload") that the flag applies to gets only

**Comment by motorman-ibm** [comment]
This command in spark
```
partdf.write.parquet("s3a://spark-s100/part.parquet")
```
receives this error
```
org.apache.hadoop.fs.s3a.AWSClientIOException: PUT 0-byte object  on part.parquet/_temporary/0/: com.amazonaws.SdkClientException: Unable to verify integrity of data upload. Client calculated content hash (contentMD5: 1B2M2Y8AsgTpgAmY7PhCfg== in base 64) didn't match hash (etag: b07dfb882ae1294301e79f8e101da980 in hex) calculated by Amazon S3.  
```

In order to get here I had to make a few code changes to progress this far
- drop ending /  in a few commands
- add mkdirs to create paths in put_object
With those, spark could make it far enough to issue the put command that received the error
```
PUT

application/octet-stream
Fri, 30 Oct 2020 21:48:26 GMT
/spark-s100/part.parquet/_temporary/0/
```

**Comment by motorman-ibm** [comment]
https://ibm.box.com/s/ty15l5usjttjnp6gp5p4i8xwtc0o1nop
has more details

**Comment by nimrod-becker** [comment]
md5 support for NS fs Fixed with #6651
